src/utils/xmlfile.cpp

```diff
@@ -446,13 +446,16 @@ unsigned long XMLfile::query(std::list<Node>& nodeselection, const std::string&
 				if (! attrname.empty())
 				{ // search for attribute node
 					attr=ele->first_attribute(attrname.c_str());
-					nodepath.append("@");
-					nodepath.append(attrname);
-					nodeselection.push_back(Node(attr,nodepath));
+					if(0 != attr) {
```

Comment:
Here do you mean:
```suggestion
					if(NULL != attr) {
```
as in if the line above did not change the default initialization (line 382) / returned an error value?
Also, what does this case mean? Is there a node, but is it invalid? Is this defined behavior?